site/content/entry/accurate-audio-emulation.md

```diff
@@ -0,0 +1,199 @@
+The fear of failure after spending so much time and effort on this project really sunk in, and [wwylele](https://github.com/wwylele) just stopped working on Teakra.
+By the middle of September, [wwylele](https://github.com/wwylele) had just had enough, and decided to work on something else in the mean time.
+
+{pict of convo about dsp status}
+
+It turns out developers are still just regular human beings.
+
+## Success... Finally!
+
+The end of November rolls around with no news about DSP LLE, and [wwylele](https://github.com/wwylele) is working tirelessly to fix many other core issues, including [fixing issues with how citra handled memory that prevented many cheat codes from working properly](https://github.com/citra-emu/citra/pull/4392) (but more on that in another blog post!)
+Out of nowhere, there's new progress on DMA.
+[wwylele](https://github.com/wwylele) worked out some extra details about the different patterns that the DMA can use to copy memory and things started taking off from there.
+The next day, he noticed an [oversight](https://github.com/wwylele/teakra/commit/d22c2d99d3b0858eb190bf9298815fcac237793a#diff-fe5ba875bf987b9df6685326294fe86fR13) in his simple DMA approach, which ended up causing the DMA to copy data to the *program* region in memory instead of the data region!
+Now that data was going to the right spot, it was much easier to debug.
+
+After some more reverse engineering and hardware testing, wwylele had a good idea of how to recreate DMA completely now, and tried it out on a custom application that produces just a simple sine wave.
+
+{audioclip of dsp fuzzy sine wave}
+
+After some more tinkering, and with renewed enthusiasm, he started fixing minor bugs here and there.
+Small oversights, he noticed a simple copy paste error in the interpreter.
+All of the sudden, everything just started working.
+
+{kirby audio}
+
+And with a few more adjustments...
+
+{more differenter audio. i dunno i'm sleepy}
+
+And finally, on Dec 6th, this is what happened after we hooked it into Citra's audio framework
+
+{playing aac audio}
+
+## The Future of Audio in Citra
+
+So what is going to happen to the current audio code?
+As you can see, there's a lot of benefit in both approaches, and we feel they both have a bright future ahead of them.
+HLE audio will almost always remain the preffered choice for users, as its much faster and produces very good audio quality, while LLE audio will be an excellent tool for finding and debugging audio issues, as after any last remaining bugs are worked out, it should have near perfect compatibility.
+With LLE audio released now, we can use it to take a deeper look at the communication between 3DS and DSP, allowing us to mimic responses from the DSP, and begin researching what each of the bytes of data actually mean.
+We plan to fix any outstanding issues with HLE audio and support all of the games, and we also plan to speed up the LLE audio so it will be usable at a good framerate!
```

Comment:
nit:
>We plan to fix any outstanding issues with the HLE audio, to support all the games, and also plan to optimize and improve the performance of LLE audio so that it will be usable at a good framerate!